src/snowflake/snowpark/functions.py

```diff
@@ -7198,7 +7203,13 @@ def udf(
         [Row(MINUS_ONE(10)=9)]
 
     """
-    session = session or snowflake.snowpark.session._get_active_session()
+    # If the local sandbox flag is true, then we should not use any session that the user may already have or pass to the decorator.
+    if snowflake.snowpark.context._is_execution_environment_sandboxed:
+        session = Session.builder.config("local_testing", True).create()
+    else:
```

Comment:
IMHO that's the wrong place to do this, and could cause unexpected behaviour. Shouldn't we instead ensure that the active session is still a singleton, but doesn't have a valid connection when sandboxing is enabled? That's a lot more future-proof as well.